packages/salesforce-adapter/test/change_validators/unknown_users.test.ts

```diff
@@ -0,0 +1,191 @@
+      const changeErrors = await validator([change])
+      expect(changeErrors).toBeEmpty()
+    })
+  })
+  describe('when a username does not exist in Salesforce', () => {
+    let change: Change
+    let validator: ChangeValidator
+    beforeEach(() => {
+      const { client } = mockAdapter({})
+      validator = changeValidator(client)
+      const beforeRecord = createInstanceElement({ fullName: 'someName', defaultCaseUser: 'someUsername' }, caseSettingsType)
+      const afterRecord = beforeRecord.clone()
+      afterRecord.value.defaultCaseUser = 'someOtherUsername'
+      change = toChange({ before: beforeRecord, after: afterRecord })
+
+      setupClientMock(client, ['someUsername'])
+    })
+
+    it('should fail validation', async () => {
+      const changeErrors = await validator([change])
+      expect(changeErrors).toHaveLength(1)
+      expect(changeErrors[0].detailedMessage).toContain('defaultCaseUser')
+      expect(changeErrors[0].detailedMessage).toContain('someOtherUsername')
+      expect(changeErrors[0].detailedMessage).toContain('someName')
+      expect(changeErrors[0].message).toContain('doesn\'t exist')
+      expect(changeErrors[0].message).toContain('someOtherUsername')
+    })
+
+    it('should pass validation if the defaultCaseOwnerType is not "User"', async () => {
+      (getChangeData(change) as InstanceElement).value.defaultCaseOwnerType = 'Other';
+      (getChangeData(change) as InstanceElement).value.defaultCaseOwner = 'BlahBlah';
+      (getChangeData(change) as InstanceElement).value.defaultCaseUser = undefined
+      const changeErrors = await validator([change])
+      expect(changeErrors).toBeEmpty()
+    })
+  })
+  describe('when an instance references multiple usernames', () => {
+    let client: SalesforceClient
+    let change: Change
+    let validator: ChangeValidator
+    beforeEach(() => {
+      client = mockAdapter({}).client
+      validator = changeValidator(client)
+      const beforeRecord = createInstanceElement({
+        fullName: 'someName',
+        defaultCaseUser: 'someUsername',
+        defaultCaseOwnerType: 'User',
+        defaultCaseOwner: 'someUsername',
+      },
+      caseSettingsType)
+      const afterRecord = beforeRecord.clone()
+      afterRecord.value.defaultCaseUser = 'someOtherUsername'
+      change = toChange({ before: beforeRecord, after: afterRecord })
+    })
+
+    it('should pass validation if both usernames exist in Salesforce', async () => {
+      setupClientMock(client, ['someUsername', 'someOtherUsername'])
+      const changeErrors = await validator([change])
+      expect(changeErrors).toBeEmpty()
+    })
+
+    it('should fail validation if only one of the usernames exists in Salesforce', async () => {
+      setupClientMock(client, ['someUsername'])
+      const changeErrors = await validator([change])
+      expect(changeErrors).toHaveLength(1)
+      expect(changeErrors[0].detailedMessage).toContain('defaultCaseUser')
```

Comment:
Why do we need this assertions? Validating there was an error is enough. I wouldnt test the content :)

Reply:
Consider this assertion instead:
```suggestion
     expect(changeErrors).toIncludeAllPartialMembers([{
        elemID: getChangeData(change).elemID,
      }])
```

Reply:
Agreed. I kept the check for the specific username in the last test because it actually checks the case where we generate two errors for the same element.